app/services/foreman/render_templates_from_folder.rb

```diff
@@ -0,0 +1,143 @@
+module Foreman
+  class RenderTemplatesFromFolder
+    attr_reader :path
+    attr_accessor :errors
+
+    def self.instance(path:)
+      @instances ||= {}
+      @instances[path] ||= new(path: path)
+    end
+
+    def initialize(path:)
+      @path = path
+      @errors = {}
+    end
+
+    def render_all
+      self.errors = {}
+      templates.each do |template|
+        next if template.snippet
+        next if template.operatingsystems.empty?
+        begin
+          render_template(template)
+        rescue StandardError => e
+          self.errors[template.name] = e.message
+        end
+      end
+      errors.empty?
+    end
+
+    def render_template(template)
+      source = source_klass.new(template)
+      scope = Foreman::Renderer.get_scope(host: host(template), source: source)
+      Foreman::Renderer.render(source, scope)
+    end
+
+    def templates
+      @templates ||= files.map { |filepath| load_file(filepath) }
+    end
+
+    def snippets
+      templates.select { |template| template.snippet? }
+    end
+
+    def load_file(filepath)
+      content = File.read(filepath)
+      metadata = Template.parse_metadata(content).deep_symbolize_keys
+      name = metadata[:name]
+      raise "Could not read name from metadata in template #{filepath}." unless name.present?
+      kind = metadata[:kind]
+      snippet = (kind == 'snippet') || !!metadata[:snippet]
+      oses = metadata[:oses] || ['RedHat']
```

Comment:
IMHO we shouldn't assume any OS.
```suggestion
      oses = metadata[:oses] || []
```